integration/spark/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -1009,6 +1009,10 @@ object CommonUtil {
 
   private def convertSparkComplexTypeToCarbonObject(data: AnyRef,
       objectDataType: DataType): AnyRef = {
+    if (data == null && (objectDataType.isInstanceOf[ArrayType]
```

Comment:
check with  objectDataType.isComplexType

Reply:
DataType is a spark class here. It doesn't have isComplexType method.

Reply:
ok